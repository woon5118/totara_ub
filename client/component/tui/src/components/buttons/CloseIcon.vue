<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Simon Chester <simon.chester@totaralearning.com>
  @module tui
-->

<template>
  <ButtonIcon
    :aria-label="ariaLabel || $str('closebuttontitle', 'core')"
    :styleclass="{ transparent: true }"
    @click.prevent="$emit('click', $event)"
  >
    <CloseIcon :size="size" />
  </ButtonIcon>
</template>

<script>
import ButtonIcon from 'tui/components/buttons/ButtonIcon';
import CloseIcon from 'tui/components/icons/Close';

export default {
  components: {
    ButtonIcon,
    CloseIcon,
  },

  props: {
    ariaLabel: String,
    size: {
      type: Number,
      default: 200,
    },
  },
};
</script>

<lang-strings>
{
  "core": [
    "closebuttontitle"
  ]
}
</lang-strings>
