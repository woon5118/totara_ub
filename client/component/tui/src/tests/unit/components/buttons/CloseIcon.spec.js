/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Simon Chester <simon.chester@totaralearning.com>
 * @module tui
 */

import { shallowMount } from '@vue/test-utils';
import CloseIcon from 'tui/components/buttons/CloseIcon';
const mocks = {
  $str: (x, y) => `[[${x}, ${y}]]`,
};

describe('CloseIcon', () => {
  it('allows passing a different label', () => {
    const wrapper = shallowMount(CloseIcon, {
      mocks,
      propsData: { ariaLabel: 'DESTROY' },
    });
    expect(wrapper.find('buttonicon-stub').props().ariaLabel).toBe('DESTROY');
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(CloseIcon, { mocks });
    expect(wrapper.element).toMatchSnapshot();
  });
});
