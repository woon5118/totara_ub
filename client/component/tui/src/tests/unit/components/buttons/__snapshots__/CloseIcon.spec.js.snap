// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon matches snapshot 1`] = `
<buttonicon-stub
  arialabel="[[closebuttontitle, core]]"
  styleclass="[object Object]"
  type="button"
>
  <closeicon-stub
    size="200"
  />
</buttonicon-stub>
`;
